propertyestimator/protocols/forcefield.py

```diff
@@ -43,54 +43,34 @@ class WaterModel(Enum):
         """
         TIP3P = 'TIP3P'
 
-    @protocol_input(str)
-    def force_field_path(self, value):
-        """The file path to the force field parameters to assign to the system.
-        This path **must** point to a json serialized `SmirnoffForceFieldSource` object."""
-        pass
-
-    @protocol_input(str)
-    def coordinate_file_path(self, value):
-        """The file path to the coordinate file which defines the system to which the
-        force field parameters will be assigned."""
-        pass
-
-    @protocol_input(Substance)
-    def substance(self):
-        """The composition of the system."""
-        pass
-
-    @protocol_output(str)
-    def system_path(self):
-        """The file path to the system object which contains the
-        applied parameters."""
-        pass
-
-    @protocol_input(WaterModel)
-    def water_model(self):
-        """The water model to apply, if any water molecules
-        are present.
-
-        Warnings
-        --------
-        This is only a temporary addition until full water model support
-        is introduced.
-        """
-        pass
-
-    def __init__(self, protocol_id):
-        """Constructs a new BaseBuildSystemProtocol object.
-        """
-        super().__init__(protocol_id)
-
-        # Inputs
-        self._force_field_path = None
-        self._coordinate_file_path = None
-        self._substance = None
-        self._water_model = BaseBuildSystemProtocol.WaterModel.TIP3P
-
-        # Outputs
-        self._system_path = None
+    force_field_path = protocol_input(
+        docstring='The file path to the force field parameters to assign to the system.',
+        type_hint=str,
+        default_value=UNDEFINED
```

Comment:
I like `UNDEFINED` vs. `None` distinction. This has bitten me before, and I may steal the convention.